src/containers/SecretsModal/SecretsModal.js

```diff
@@ -186,10 +187,10 @@ export /* istanbul ignore next */ class SecretsModal extends Component {
       annotationsObject[key] = value;
     }
     postData.metadata.annotations = annotationsObject;
-
     if (Object.keys(invalidFields).length === 0) {
-      this.props.createSecret(postData, namespace);
-      this.props.handleNew();
+      await this.props.createSecret(postData, namespace).then(result => {
```

Comment:
Should we also be waiting for the `handleCreateSecret` here? Combining async/await and regular promise usage in the same statement can be confusing. Sticking to a single style makes it clearer what the intention is, and can help to avoid subtle errors.

Reply:
Yep good point - updated to await `handleCreateSecret` as well

Reply:
Not quite what I meant, you still have async/await and promise.then mixed here. Rewriting to use just async/await you'd have:

```js
const result = await this.props.createSecret(postData, namespace);
await this.props.handleCreateSecret(result);
```

Reply:
ah ok, got it :)